src/bin/enclave-runc-wrapper.rs

```diff
@@ -0,0 +1,241 @@
+                    .unwrap();
+                let p = std::path::PathBuf::from(log_directory);
+                let file_name = p.file_name().unwrap().to_str().unwrap();
+                let mut splitter = file_name.split('_');
+                let namespace = splitter.next().unwrap().to_string();
+
+                return Ok(Some(namespace));
+            } else if annotations.contains_key(ANNOTATION_CONTAINERD_SANDBOX_ID) {
+                // When a container is running as a part of a previously created
+                // pod, the log directory path has to be retrieved from the
+                // sandbox container.
+                let sandbox_id = annotations[ANNOTATION_CONTAINERD_SANDBOX_ID]
+                    .as_str()
+                    .unwrap();
+                return container_namespace(sandbox_id, container_root);
+            }
+            Ok(None)
+        }
+        None => Ok(None),
+    }
+}
+
+async fn policy_label(
+    namespace_o: Option<std::string::String>,
+) -> Result<enclave::ContainerPolicyLevel, kube::Error> {
+    let namespace_s: std::string::String;
+
+    match namespace_o {
+        Some(v) => namespace_s = v,
+        None => return Ok(enclave::ContainerPolicyLevel::Baseline),
+    }
+
+    // Apply the privileged policy for kube-system containers immediately.
+    // Otherwise the core k8s components (apiserver, scheduler) won't be able
+    // to run.
+    if namespace_s.as_str() == "kube-system" {
+        return Ok(enclave::ContainerPolicyLevel::Privileged);
+    }
+
+    let kubeconfig =
+        kube::config::Kubeconfig::read_from(std::path::Path::new("/etc/kubernetes/admin.conf"))?;
+    let options = kube::config::KubeConfigOptions::default();
+    let config = kube::config::Config::from_custom_kubeconfig(kubeconfig, &options).await?;
+    let client = kube::Client::try_from(config)?;
+
+    let namespaces: kube::api::Api<v1::Namespace> = kube::api::Api::all(client);
+    let namespace = namespaces.get(&namespace_s).await?;
+
+    match namespace.metadata.labels.get(LABEL_POLICY_ENFORCE) {
+        Some(s) => match &s[..] {
+            "restricted" => Ok(enclave::ContainerPolicyLevel::Restricted),
+            "baseline" => Ok(enclave::ContainerPolicyLevel::Baseline),
+            "privileged" => Ok(enclave::ContainerPolicyLevel::Privileged),
+            _ => Ok(enclave::ContainerPolicyLevel::Baseline),
+        },
+        None => Ok(enclave::ContainerPolicyLevel::Baseline),
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let pid = nix::unistd::getpid();
+    let pid_u = u32::try_from(libc::pid_t::from(pid))?;
+
+    let mut skip_arg = false;
+    let mut root_arg = false;
+    let mut container_id_arg = false;
+
+    let mut add_container = false;
+    let mut delete_container = false;
+    let mut start_container = false;
+
+    let mut container_root: Option<std::string::String> = None;
+    let mut container_id_o: Option<std::string::String> = None;
```

Comment:
Perhaps a struct definition without any new dependencies or [clap-rs](https://github.com/clap-rs/clap) might be suitable for this?

Reply:
I was thinking about using clap-rs, but it would require me to write down literally all subcommands and options that runc supports, which could be an overkill and we would have to follow all the CLI changes in runc, possibly with supporting some particular versions.

My assumption here is that I'm defining a subset of subcommand which have `container-id` as their first positional argument. The list of those subcommands we're interested in wrapping will likely not change, as well as the fact that `container-id` is their 1st pos argument.

Reply:
Ah I see. Thanks for the explanation.